content/blog/2021-04-19-experiment-refs.md

```diff
@@ -0,0 +1,277 @@
+learning (ML) pipelines and experiments using Git commits. While this works very
+well for versioning mature projects and models, for projects under active
+development that may require generating hundreds of experiments or more in a
+single day, typical Git workflows can be difficult to work with. This type of
+rapid experimentation may appear to fit nicely with the concept of Git feature
+branches, but a Git repository with such large numbers of branches will
+eventually become too unwieldy to manage.
+
+In DVC 2.0, we’ve introduced a new feature set aimed at simplifying the
+versioning of lightweight ML experiments. DVC now provides a series of `dvc exp`
+commands which allow you to easily generate new experiments with modified
+hyperparameters, and to quickly compare their results. In this post, we’ll show
+how DVC leverages the power of Git references to track each experiment, while
+also completely abstracting away the need for you to manually manage a
+potentially unlimited number of Git feature branches or tags.
+
+_Note: This post mainly focuses on the “How?” side of DVC 2.0 experiments. For a
+great overview of the “What?” check out our
+[2.0 release post](/blog/dvc-2-0-release) and our
+[Get Started: Experiments](/doc/start/experiments) guide._
+
+## Experiments in DVC 2.0
+
+At the heart of the new experiments feature is the `dvc exp run` command.
+Whenever a pipeline is executed with `dvc exp run`, the results will be
+automatically tracked by DVC as a single “experiment”. DVC will track everything
+in your workspace as a part of the experiment, including unstaged changes made
+prior to execution. This means that DVC experiments can be used to test the
+result of changes to DVC-tracked data or pipeline parameters, as well as changes
+to Git-tracked code.
+
+![Example experiment run](/uploads/images/2021-04-19/exp-run.gif 'Example experiment run')
+
+_Note: You can follow along with the commands used in this example and
+throughout this post, using our
+[example-get-started](https://github.com/iterative/example-get-started)
+repository._
+
+Now let’s take a deeper look into what actually happened when we ran our
+experiment. Starting from the latest commit in our repository’s `master` branch,
+we invoked `dvc exp run --set-param` to generate a new experiment with the
+specified parameter value. DVC then reproduced our pipeline as if we had
+manually edited our `params.yaml` to contain that parameter change (setting
+`featurize.max_features` to `2000`), and then saved the results in a new
+experiment named `exp-9b1a5`.
+
+_Note: `--temp` is an optional flag for `dvc exp run` which makes DVC run the
+experiment in a temporary directory instead of in our main workspace. We use
+`--temp` here to illustrate how DVC tracks experiment changes without needing to
+modify the workspace at all, but experiments can also be run directly in your
+workspace as well (like in traditional `dvc repro` workflows)._
```

Comment:
Minor: I don't see the value of running it in temp dir. We can keep it is as.

Reply:
The main reasoning was that some extraneous internal DVC-only refs will show up in the low-level git commands if we demo with workspace runs. These refs are kept in the execution environment only, so for `--temp` runs they would not show up in our demo commands.

But I don't think it's a big deal either way, so I can adjust the demo to do workspace runs